server/asset/btc/rpcclient.go

```diff
@@ -249,6 +284,22 @@ func (rc *RPCClient) GetBlockVerbose(blockHash *chainhash.Hash) (*GetBlockVerbos
 	return res, rc.call(methodGetBlock, anylist{blockHash.String(), arg}, res)
 }
 
+// GetBlockHash fetches the block hash for the block at the given index.
+func (rc *RPCClient) GetBlockHash(index int64) (*chainhash.Hash, error) {
+	var blockHashStr string
+	err := rc.Call(methodGetBlockHash, []interface{}{index}, &blockHashStr)
+	if err != nil {
+		return nil, err
+	}
+
+	blockHash, err := chainhash.NewHashFromStr(blockHashStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockHash, nil
```

Comment:
The boiler plate isn't really needed here. `return chainhash.NewHashFromStr(blockHashStr)`.

Actually, this whole function could be `return rc.callHashGetter(methodGetBlockHash, anylist{index})`, but I can't stand that `callHashGetter` helper TBH.  :laughing:

Reply:
I could leave it all in `checkTxIndex` but I'd like to keep it cleaner.